geode-core/src/main/java/org/apache/geode/internal/cache/tier/sockets/AcceptorImpl.java

```diff
@@ -983,7 +994,11 @@ public void runSelectorLoop() {
             break;
           }
           ServerConnection sc;
-          registeredKeys = checkRegisteredKeys(registeredKeys);
+          long delta = System.currentTimeMillis() - lastCheckedTime;
+          if (delta >= checkRegisteredKeysInterval) {
```

Comment:
I think this needs an additional check just in case `System.currentTimeMillis()` returns the same value.

`if (checkRegisteredKeysInterval == 0 || ...)`

Reply:
requested change was done